src/extensions/mvs/helpers/utils.ts

```diff
@@ -0,0 +1,317 @@
+    return result as Omit<T, K>;
+}
+
+/** Create an object from keys and values (first key maps to first value etc.) */
+export function objectFromKeysAndValues<K extends keyof any, V>(keys: K[], values: V[]): Record<K, V> {
+    const obj: Partial<Record<K, V>> = {};
+    for (let i = 0; i < keys.length; i++) {
+        obj[keys[i]] = values[i];
+    }
+    return obj as Record<K, V>;
+}
+
+/** Equivalent to Pythonic `{k: getValue(k) for k in array}` */
+export function mapArrToObj<K extends keyof any, V>(array: readonly K[], getValue: (key: K) => V): Record<K, V> {
+    const result = {} as Record<K, V>;
+    for (const key of array) {
+        result[key] = getValue(key);
+    }
+    return result;
+}
+
+/** Equivalent to Pythonic `{k: getValue(k, v) for k, v in obj.items()}` */
+export function mapObjToObj<K extends keyof any, VIn, VOut>(obj: Record<K, VIn>, getValue: (key: K, value: VIn) => VOut): Record<K, VOut> {
+    const result = {} as Record<K, VOut>;
+    for (const key in obj) {
+        result[key] = getValue(key, obj[key]);
+    }
+    return result;
+}
+
+/** Decide if `obj` is a good old object (not array or null or other type). */
+export function isReallyObject(obj: any): boolean {
+    return typeof obj === 'object' && obj !== null && !Array.isArray(obj);
+}
+
+/** Return a copy of object `obj` with sorted keys and dropped keys whose value is undefined. */
+export function sortObjectKeys<T extends {}>(obj: T): T {
+    const result = {} as T;
+    for (const key of Object.keys(obj).sort() as (keyof T)[]) {
+        const value = obj[key];
+        if (value !== undefined) {
+            result[key] = value;
+        }
+    }
+    return result;
+}
+
+/** Like `Promise.all` but with objects instead of arrays */
+export async function promiseAllObj<T extends {}>(promisesObj: { [key in keyof T]: Promise<T[key]> }): Promise<T> {
+    const keys = Object.keys(promisesObj);
+    const promises = Object.values(promisesObj);
+    const results = await Promise.all(promises);
+    return objectFromKeysAndValues(keys, results) as any;
+}
+
+
+/** Return an array containing integers from [start, end) if `end` is given,
+ * or from [0, start) if `end` is omitted. */
+export function range(start: number, end?: number): number[] {
+    if (end === undefined) {
+        end = start;
+        start = 0;
+    }
+    const length = Math.max(end - start, 0);
+    const result = Array(length);
+    for (let i = 0; i < length; i++) {
+        result[i] = start + i;
+    }
+    return result;
+}
+
+/** Copy all elements from `src` to the end of `dst`.
+ * Equivalent to `dst.push(...src)`, but avoids storing element on call stack. Faster that `extend` from Underscore.js.
+ * `extend(a, a)` will double the array
+ */
+export function extend<T>(dst: T[], src: ArrayLike<T>): void {
+    const offset = dst.length;
+    const nCopy = src.length;
+    dst.length += nCopy;
+    for (let i = 0; i < nCopy; i++) {
+        dst[offset + i] = src[i];
+    }
```

Comment:
Just use `.push` here.

Reply:
This is to avoid resizing `dst` multiple times when `src.length >> dst.length`